.github/workflows/wheels-build.yml

```
@@ -0,0 +1,166 @@
+name: Build RAPIDS wheels
+
+on:
+  workflow_call:
+    inputs:
+      # repo and branch
+      repo:
+        type: string
+      branch:
+        type: string
+      date:
+        type: string
+      sha:
+        type: string
+      build_type:
+        required: true
+        type: string
+      build_script:
+        type: string
+        default: "ci/build_wheel.sh"
+
+      # allow a bigger runner instance
+      node_type:
+        required: false
+        type: string
+        default: "cpu16"
+
+      # general settings
+      matrix_filter:
+        type: string
+        default: "."
+
+      # Extra repository that will be cloned into the project directory.
+      extra-repo:
+        required: false
+        type: string
+        default: ''
+      extra-repo-sha:
+        required: false
+        type: string
+        default: ''
+      # Note that this is the _name_ of a secret containing the key, not the key itself.
+      extra-repo-deploy-key:
+        required: false
+        type: string
+        default: ''
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  actions: read
+  checks: none
+  contents: read
+  deployments: none
+  discussions: none
+  id-token: write
+  issues: none
+  packages: read
+  pages: none
+  pull-requests: read
+  repository-projects: none
+  security-events: none
+  statuses: none
+
+jobs:
+  wheel-epoch-timestamp:
+    name: wheel epoch timestamper
+    runs-on: ubuntu-latest
+    outputs:
+      rapids_epoch_timestamp: ${{ steps.get_epoch_time.outputs.rapids_epoch_timestamp }}
+
+    steps:
+    - name: Get epoch timestamp to use as nightly version
+      id: get_epoch_time
+      run: echo "RAPIDS_EPOCH_TIMESTAMP=$(date +%s)" >> "${GITHUB_OUTPUT}"
+
+  compute-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      MATRIX: ${{ steps.compute-matrix.outputs.MATRIX }}
+    steps:
+      - name: Compute Build Matrix
+        id: compute-matrix
+        run: |
+          set -eo pipefail
+
+          export MATRIX="
+          - { ctk: '11.8.0', arch: 'amd64', python: '3.9', linux_ver: 'centos7' }
+          - { ctk: '11.8.0', arch: 'amd64', python: '3.10', linux_ver: 'centos7' }
+          - { ctk: '11.8.0', arch: 'arm64', python: '3.9',  linux_ver: 'ubuntu20.04' }
+          - { ctk: '11.8.0', arch: 'arm64', python: '3.10', linux_ver: 'ubuntu20.04' }
+          - { ctk: '12.0.1', arch: 'amd64', python: '3.9',  linux_ver: 'centos7' }
+          - { ctk: '12.0.1', arch: 'amd64', python: '3.10',  linux_ver: 'centos7' }
+          - { ctk: '12.0.1', arch: 'arm64', python: '3.9', linux_ver: 'ubuntu20.04' }
+          - { ctk: '12.0.1', arch: 'arm64', python: '3.10', linux_ver: 'ubuntu20.04' }
+          "
+
+          echo "MATRIX=$(
+            yq -n -o json 'env(MATRIX)' | \
+            jq -c '${{ inputs.matrix_filter }} | {include: .}' \
+          )" | tee --append "${GITHUB_OUTPUT}"
+
+  wheel-build:
```

```suggestion
  build:
```

In the context of a `wheels-build.yml` workflow, a job named `wheel-build` might be a bit redundant.

In other other workflows, we simply use `build` (see [here](https://github.com/rapidsai/shared-action-workflows/blob/614043874f8b8998e68a903db73bfb4253f78c3d/.github/workflows/conda-python-build.yaml#L68)) for succinctness.